src/garage/tf/algos/_rl2trpo.py

```diff
@@ -0,0 +1,117 @@
+"""Trust Region Policy Optimization."""
+from garage.tf.algos._rl2npo import RL2NPO
+from garage.tf.optimizers import ConjugateGradientOptimizer
+from garage.tf.optimizers import PenaltyLbfgsOptimizer
+
+
+class RL2TRPO(RL2NPO):
```

Comment:
there is not much reason to use the names `RL2PPO` and RL2TRPO` for these because they are very simple and the user won't interact with them.

i think it would be better for RL2TRPO and RL2PPO to be subclasses of RL2 which allow the user to pass PPO/TRPO constructor args directly rather than using `inner_algo_args`. You can replace what you currently call RL2TRPO and RL2PPO which just different calls to RL2NPO.

Basically (I use *args for brevity, but please expand all of the args per usual)
```python
class RL2NPO(NPO):  # private, unchanged
      # ...


class RL2(MetaRLAlgorithm):  # public
    
    def __init__(self, *rl2_stuff, inner_algo, inner_algo_args):
        # ...
        if inner_algo == 'ppo'
            optimizer = FirstOrderOptimizer
            # etc...
        elif inner_algo == 'trpo'
            optimizer = ConjugateGradientOptimizer
            # etc...
        else:
            raise ValueError('inner_algo must be one of \'ppo\' or \'trpo\', but got \'{}\' instead'.format(inner_algo))
        
        self._inner_algo = RL2NPO(optimizer=optimizer, *inner_algo_args)


class RL2TRPO(RL2):   # public

    def __init__(self, *rl2_stuff, *trpo_stuff):
        # ...
        super.__init__(*rl2_stuff, inner_algo='trpo', inner_algo_args=trpo_stuff)


class RL2PPO(RL2):  # public

    def __init__(self, *rl2_stuff, *ppo_stuff):
        # ...
        super.__init__(*rl2_stuff, inner_algo='ppo', inner_algo_args=ppo_stuff)

Reply:
Should we expand `*rl2_stuff` in `RL2`? I think arguments for `PPO` and `TRPO` are not identifical, e.g. only TRPO has `kl_constraint`

Reply:
Actually `RL2` should be an ABC?

Reply:
RL2 should not be an ABC, because someone could provide an inner_algo and instantiate it. we just won't advertise it.

note that in my example, RL2 has `inner_algo_args` (a dict, just like now) and `RL2PPO`/`RL2TRPO` has `*ppo_stuff`/`trpo_stuff` (i.e. expanded). Because RL2PPO is just like PPO, it should have all the PPO args expanded.

Reply:
so basically, no, RL2's signature doesn't change

Reply:
In your example, logic about inner algorithm is done within RL2, while in `PPO` and `TRPO`, the logic is done in the constructor, i.e. before calling `super.init()`. To stick with garage API, I think it's better to make RL2 an ABC, and if one wants to add a new inner algorithm say `RL2XXX`, he would subclass RL2 and handle the logic, and we don't have to touch RL2 ever.

Reply:
@krzentner PTAL